core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -109,6 +111,9 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[HttpBroadcast[_]], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
+    kryo.register(classOf[TaskMetrics], new KryoJavaSerializer())
+    kryo.register(classOf[DirectTaskResult[_]], new KryoJavaSerializer())
+    kryo.register(classOf[IndirectTaskResult[_]], new KryoJavaSerializer())
```

Comment:
bq. people may forget to register new classes if they just add an Option field to TaskMetrics in future

Addition to TaskMetrics would be reviewed, right ?
A comment can be added to TaskMetrics reminding them to register corresponding class.